### PR TITLE
Deprecate declaration-block-properties-order

### LINKED ISSUE #2010: Deprecate declaration-block-properties-order in favour of more specific community plugins
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I want to test my Sass (scss) without having the properties being in a set order.

> Which rule, if any, is this issue related to?

declaration-block-properties-order

> What stylelint configuration is needed to reproduce this issue?

``` json
{
  "rules": {
    "declaration-block-properties-order": [
      {
        order: "flexible",
        properties: [
          "position",
          "top",
          "right",
          "bottom",
          "left"
        ]
      },
      {
        order: "flexible",
        properties: [
          "flex",
          "flex-direction",
          "flex-order",
          "flex-pack",
          "flex-align"
        ]
      },
      {
        order: "flexible",
        properties: [
          "margin",
          "margin-top",
          "margin-right",
          "margin-bottom",
          "margin-left"
        ]
      },
      {
        order: "flexible",
        properties: [
          "font",
          "font-family",
          "font-size",
          "font-style",
          "font-weight"
        ]
      },
      {
        order: "flexible",
        properties: [
          "background",
          "background-color",
          "background-image",
          "background-repeat",
          "background-attachment",
          "background-position",
          "background-position-x",
          "background-position-y",
          "background-clip",
          "background-origin",
          "background-size",
        ]
      },
      {
        order: "flexible",
        properties: [
          "transition",
          "transition-delay",
          "transition-timing-function",
          "transition-duration",
          "transition-property",
        ]
      },
      {
        order: "flexible",
        properties: [
          "animation",
          "animation-name",
          "animation-duration",
          "animation-fill-mode",
          "animation-play-state",
          "animation-timing-function",
          "animation-delay",
          "animation-iteration-count",
          "animation-direction"
        ]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

"gulp-stylelint": "3.4.0",

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

Code to flag, but with the order in each properties: [...] above (not in the listed order).

> What actually happened (e.g. what warnings or errors you are getting)?

Flags loads of errors, because the code is not written in the order of each block above.
## 

In an ideal world, I want the code to check through and validate the code. However I can't expect all my development team to remember the order.

E.G. must use background before transition but only after the flex code. As an example I don't mind where the flex and positioning are placed however when someone codes positioning. I want it to go:

Pass linting:

``` sass
position: absolute;
top: 0;
right: 0;
bottom:
left: 0;
```

Flag in NPM:

``` sass
right: 0;
top: 0;
bottom:
position: absolute;
left: 0;
```

Many thanks.

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #2010

### CODE REVIEW

File: docs/user-guide/plugins.md [hunk 1]
```diff
@@ -4,30 +4,18 @@ Plugins are rules and sets of rules built by the community that support methodol
 
 ## Code style
 
--   [`stylelint-declaration-block-order`](https://github.com/hudochenkov/stylelint-declaration-block-order): Specify the order of groups of at-rules, nested rules, custom properties, declarations and dollar variables in declaration blocks.
--   [`stylelint-property-groups-structure`](https://github.com/hudochenkov/stylelint-property-groups-structure): Require or disallow an empty line before user-defined property groups and specify the order of the properties within these groups (incorporates the `declaration-block-properties-order` rule).
+-   [`stylelint-order](https://github.com/hudochenkov/stylelint-order): A plugin pack of rules for specifying the ordering of things e.g. properties within declaration blocks.
```
Comment (hudochenkov) [inline docs/user-guide/plugins.md]: Syntax error here.

**Comment by davidtheclark** [comment]
I think we need a deprecation error in the code. People will not be checking up on this text, so if we don't add one in the code the removal will still come as a surprise.

**Comment by jeddy3** [comment]
> I think we need a deprecation error in the code. 

Agreed. I've reverted the changelog change, so we can do this in another issue.

This PR is now just about updating the release planning and tidying up the plugin list.

**Comment by davidtheclark** [comment]
👍

**Pull request merged**